hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/HoodieMergeHelper.java

```diff
@@ -123,8 +123,12 @@ public void runMerge(HoodieTable<?, ?, ?, ?> table,
         Configuration bootstrapFileConfig = new Configuration(table.getHadoopConf());
         bootstrapFileReader =
             HoodieFileReaderFactory.getReaderFactory(recordType).getFileReader(bootstrapFileConfig, bootstrapFilePath);
-        recordIterator = new MergingIterator(baseFileRecordIterator, bootstrapFileReader.getRecordIterator(),
-            (left, right) -> left.joinWith(right, mergeHandle.getWriterSchemaWithMetaFields()));
+
+        recordIterator = new MergingIterator<>(
+            baseFileRecordIterator,
+            bootstrapFileReader.getRecordIterator(),
+            (left, right) ->
+                left.joinWith(right, mergeHandle.getWriterSchemaWithMetaFields()));
```

Comment:
should have avoid the style change here and kept it 1-line diff to make reviewer's life easier

Reply:
There was an actual change here (we're passing the schema), but this change is reverted now